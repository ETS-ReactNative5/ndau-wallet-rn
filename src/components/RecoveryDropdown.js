import React, { Component } from 'react'

import {
  StyleSheet,
  TouchableOpacity,
  Text,
  Platform,
  NativeModules
} from 'react-native'
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from 'react-native-responsive-screen'
import Autocomplete from 'react-native-autocomplete-input'
import styleConstants from '../css/styleConstants'
import AppConstants from '../AppConstants'

class RecoveryDropdown extends Component {
  constructor (props) {
    super(props)

    this.state = {
      query: null,
      list: [],
      textColor: '#ffffff'
    }

    this.retrievedData = false
  }
  // TODO: we are going to have to write them somewhere...
  onPress (title) {
    console.log(`title selected is ${title}`)
    this.setState({ query: title, list: [] })
    this.props.addToRecoveryPhrase(title)
    this.props.setDisableArrows(false)
  }

  clearWord = () => {
    this.setState({ query: '' })
  }

  goGetTheData = async query => {
    if (query && !this.retrievedData) {
      const words = await NativeModules.KeyaddrManager.keyaddrWordsFromPrefix(
        AppConstants.APP_LANGUAGE,
        query,
        5
      )
      this.retrievedData = true

      console.log(`words; ${words} query: ${query}`)
      if (words === query) {
        this.props.setDisableArrows(false)
        this.setState({ list: [] })
        return
      }
      wordsArray = words.split(' ')

      if (wordsArray[0] === '') {
        this.setState({ textColor: '#ff0000', list: [] })
        this.props.setAcquisitionError(true)
      } else {
        this.setState({ textColor: '#ffffff' })
        this.props.setAcquisitionError(false)
      }

      this.setState({
        list: wordsArray.length >= 0 ? wordsArray : []
      })
    }
  }

  getData = query => {
    this.goGetTheData(query)
    let dropdownValueIsPrinted = false
    if (
      this.state.list.length === 1 &&
      this.state.list[0] === this.state.query
    ) {
      dropdownValueIsPrinted = true
    }
    return dropdownValueIsPrinted ? [] : this.state.list
  }

  render () {
    const { query } = this.state
    const mainStyle = {
      backgroundColor: styleConstants.APP_BACKGROUND_COLOR,
      color: this.state.textColor,
      fontSize: 20,
      fontFamily: 'TitilliumWeb-Regular',
      borderRadius: 3,
      zIndex: 1
    }

    return (
      <Autocomplete
        style={mainStyle}
        autoCapitalize='none'
        autoCorrect={false}
        autoFocus
        containerStyle={styles.autocompleteContainer}
        inputContainerStyle={styles.containerStyle}
        data={this.getData(query)}
        defaultValue={query || this.props.recoveryWord}
        onChangeText={text => {
          this.retrievedData = false
          this.setState({
            query: text,
            list: text === '' ? [] : this.state.list
          })
          this.props.addToRecoveryPhrase(text)
        }}
        listContainerStyle={styles.listContainerStyle}
        placeholderTextColor='#ffffff'
        renderItem={item => (
          <TouchableOpacity onPress={() => this.onPress(item)}>
            <Text style={styles.itemText}>{item}</Text>
          </TouchableOpacity>
        )}
      />
    )
  }
}

var styles = StyleSheet.create({
  container: {
    flex: 1
  },
  autocompleteContainer: {
    ...Platform.select({
      ios: {
        backgroundColor: styleConstants.APP_BACKGROUND_COLOR
      },
      android: {
        flex: 1,
        borderRadius: 3
      }
    })
  },
  inputContainerStyle: {
    color: '#ffffff',
    fontSize: 20,
    fontFamily: 'TitilliumWeb-Regular',
    backgroundColor: styleConstants.APP_BACKGROUND_COLOR
  },
  containerStyle: {
    alignContent: 'center',
    justifyContent: 'center',
    borderRadius: 3,
    width: wp('60%'),
    backgroundColor: styleConstants.APP_BACKGROUND_COLOR,
    overflow: 'hidden',
    ...Platform.select({
      ios: {
        paddingLeft: wp('2%'),
        height: hp('6%')
      },
      android: {
        height: hp('7%')
      }
    })
  },
  listContainerStyle: {
    width: wp('60%')
  },
  itemText: {
    color: '#ffffff',
    fontSize: 20,
    fontFamily: 'TitilliumWeb-Light',
    backgroundColor: styleConstants.APP_BACKGROUND_COLOR,
    paddingLeft: wp('2%'),
    paddingTop: wp('2%'),
    paddingBottom: wp('2%')
  }
})

export default RecoveryDropdown
