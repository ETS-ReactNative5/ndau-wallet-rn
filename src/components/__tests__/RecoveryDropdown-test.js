import { mount } from 'enzyme'
import React from 'react'
import { StyleSheet } from 'react-native'
import RecoveryDropdown from '../RecoveryDropdown'

import renderer from 'react-test-renderer'

describe('testing RecoveryDropdown...', () => {
  let styles = StyleSheet.create({
    wizardText: {
      color: '#ffffff',
      fontSize: 20
    }
  })
  const navigator = {
    setStyle: () => {}
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <RecoveryDropdown
          recoveryPhrase={''}
          parentStyles={styles}
          navigator={navigator}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
