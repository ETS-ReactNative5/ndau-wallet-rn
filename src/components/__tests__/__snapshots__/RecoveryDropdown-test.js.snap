// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing RecoveryDropdown... renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "zIndex": 1,
      },
      Object {
        "backgroundColor": "#1c2227",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderColor": "#b9b9b9",
          "borderRadius": 1,
          "borderWidth": 1,
        },
        Object {
          "alignContent": "center",
          "backgroundColor": "#1c2227",
          "borderRadius": 3,
          "height": 80,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingLeft": 15,
          "width": 450,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      containerStyle={
        Object {
          "backgroundColor": "#1c2227",
        }
      }
      data={Array []}
      defaultValue=""
      inputContainerStyle={
        Object {
          "alignContent": "center",
          "backgroundColor": "#1c2227",
          "borderRadius": 3,
          "height": 80,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingLeft": 15,
          "width": 450,
        }
      }
      keyboardShouldPersistTaps="always"
      listContainerStyle={
        Object {
          "width": 450,
        }
      }
      onChangeText={[Function]}
      onEndEditing={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      placeholderTextColor="#ffffff"
      renderItem={[Function]}
      renderSeparator={null}
      renderTextInput={[Function]}
      rowHasChanged={[Function]}
      style={
        Object {
          "backgroundColor": "#1c2227",
          "borderRadius": 3,
          "color": "#ffffff",
          "fontFamily": "TitilliumWeb-Regular",
          "fontSize": 20,
          "zIndex": 1,
        }
      }
    />
  </View>
  <View
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "width": 450,
      }
    }
  />
</View>
`;
